### PR TITLE
Users Phone Number Option

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?

The ability for users to add a personal phone number for the location - instead of the person booking the meeting putting their number.

_Internal: _ [Twist Chat](https://twist.com/a/195135/ch/554223/t/3442331/)

Fixes # (issue)

 Loom Video: https://www.loom.com/share/8d3bac91c70c49908f76007d7a19df8d

## Type of change
- [X] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Add the new location type to a event
- [ ] Enter your phone number
- [ ] Place a booking

## Checklist
- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 1]
```diff
@@ -66,6 +66,7 @@ type BookingFormValues = {
   locationType?: LocationType;
   guests?: string[];
   phone?: string;
+  phoneNumber?: string; // Maybe come up with a better way to name this to distingish between two types of phone numbers
```
Comment (zomars) [inline apps/web/components/booking/pages/BookingPage.tsx:69]:
```suggestion
  hostPhoneNumber?: string;
```

File: apps/web/pages/event-types/[type].tsx [hunk 2]
```diff
@@ -1866,7 +1910,10 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
                     if (newLocation === LocationType.Link) {
                       details = { link: values.locationLink };
                     }
-
+                    if (newLocation === LocationType.UserPhone) {
+                      details = { phoneNumber: values.locationPhoneNumber };
+                    }
+                    console.log({ newLocation, details });
```
Comment (zomars) [inline apps/web/pages/event-types/[type].tsx:1921]:
```suggestion
```

File: packages/app-store/utils.ts [hunk 3]
```diff
@@ -36,7 +36,8 @@ function translateLocations(locations: OptionTypeBase[], t: TFunction) {
 const defaultLocations: OptionTypeBase[] = [
   { value: LocationType.InPerson, label: "in_person_meeting" },
   { value: LocationType.Link, label: "link_meeting" },
-  { value: LocationType.Phone, label: "phone_call" },
+  { value: LocationType.Phone, label: "their_phone_number" },
```
Comment (zomars) [inline packages/app-store/utils.ts:39]:
```suggestion
  { value: LocationType.Phone, label: "attendee_phone_number" },
```

File: packages/app-store/utils.ts [hunk 4]
```diff
@@ -36,7 +36,8 @@ function translateLocations(locations: OptionTypeBase[], t: TFunction) {
...
+  { value: LocationType.UserPhone, label: "your_phone_number" },
```
Comment (zomars) [inline packages/app-store/utils.ts:40]:
```suggestion
  { value: LocationType.UserPhone, label: "host_phone_number" },
```

**Review by zomars — COMMENTED** [review]
Thank you @sean-brydon very well done. Left some areas to improve. Also I know it's besides the scope of this PR but we should:

- Format the phone number accordingly `+1 (667) 123 45 67` (lookup a lightweight library for this)
- Use `zod` to actually validate phone numbers on capture.
- Maybe add a `<a href="tel:xxxxx>` link to the email .
- Add a `Host phone number:` label since it's not very clear as it is now:
  image

**Comment by agustif** [comment]
Hey Sean! Code-wise looks good to me, but I just was testing on vercel deployment, I'm not seeing the Where? On the success page after booking?
image

Do you know what I might be missing?

File: apps/web/pages/event-types/[type].tsx [hunk 5]
```diff
@@ -407,6 +407,33 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
...
           </div>
         );
+      case LocationType.UserPhone:
+        return (
+          <div>
+            <label htmlFor="phonenumber" className="block text-sm font-medium text-gray-700">
+              {t("set_your_phone_number")}
+            </label>
+            <div className="mt-1">
+              <input
```
Comment (CarinaWolli) [inline apps/web/pages/event-types/[type].tsx:417]: Maybe we could use the already existing component for phone input here? [Link to PhoneInput component](https://github.com/calcom/cal.com/blob/feat/phone-location/apps/web/components/ui/form/PhoneInput.tsx)

**Pull request merged**